### DISCUSSION TITLE
Inquiry About Read Cache Workload Testing Practices

### DISCUSSION THREAD

**Discussion by PsiACE** [GreptimeTeam/greptimedb#3687] [Q&A] [answered]
I am currently in the process of evaluating various caching solutions and their performance characteristics, and I am particularly interested in the methodologies your team employs to test Read Cache efficiency.

Could you share some insights into the types of workloads you use to test Read Cache? I am looking for both single-node and distributed benchmarking scenarios that can help me understand how the Read Cache performs under different conditions.

Moreover, if there are any specific benchmarking tools or frameworks that you would recommend for these purposes, I would greatly appreciate learning about them.

**Comment by PsiACE** [comment]
I'm aware that metrics provide insights into hit/miss ratios, yet I'm keen to grasp the typical data sets and query patterns employed for benchmarking Greptime.  I'm also looking for workload patterns that will clearly demonstrate the impact of Read Cache on performance within these metrics.

**Comment by waynexia** [comment, marked as answer]
To be honest, read cache at present is quite naïve compared to other maturation systems. We've implemented a layered cache on file using a local disk, some memory cache for file pages, internal structures and metadata. And most of them are simple capacity-bounded LRU. Don't know if this answers your curiosity, and please ask any details you are still interested.

> I'm also looking for workload patterns that will clearly demonstrate the impact of Read Cache on performance within these metrics.

To my understanding, methods like counting miss/hit times can give real-time insight into how a cache works. For benchmarking whose workload is simpler and easy to reproduce, run it twice with and without cache respectively should provide a reliable result.

>Moreover, if there are any specific benchmarking tools or frameworks that you would recommend for these purposes, I would greatly appreciate learning about them.

We currently leverage two suites to benchmark the db: [tsbs](https://github.com/GreptimeTeam/tsbs) and [vmbench](https://github.com/VictoriaMetrics/prometheus-benchmark). And we're also considering making one to cover more scenarios. Appreciate it in advance if you have any insight on how to build a benchmark tool that can share with.
